### ISSUE TITLE
[pwrmgr] Error FSM state is not observable in pwrmgr_fsm

### ISSUE THREAD

**Issue by Razer6** [lowRISC/opentitan#19124]
## Description

The pwrmgr_fsm implements a sparsely encoded state machine. All deviations from the valid state, e.g., due to a fault on the state register, lead the FSM to transfer to the `default` state.

However, compared to other FSMs, the particular FSM does not implement a dedicated error signal for this FSM. Consequently, it is impossible to observe if the error state is entered.

## Potential Solution

Implement a dedicated error signal for the FSM that asserts if the FSM reaches the default state.

**Comment by matutem** [comment]
I assume this is for simulation observability in the RTL? Notice the RTL instantiates an assertion that fires when FSM is not in a valid state. Do you need additional indications?

**Comment by Razer6** [comment]
Yes it was was for formal fault verification. The idea here was to prove that a fault on the state machine leads the state being redirected to the default state and **not** a another _valid_ state. I used that error signal from the default state as a success condition, meaning that that particular fault could not be used to reach a harmful state.

**Comment by matutem** [comment]
In prim_sparse_fsm_flop there is a SIMULATION-only signal unused_err_o which is set to 1 if the updated state is not valid. It may be possible to use that for formal, right?

**Comment by Razer6** [comment]
Back at the time I was doing formal on a post-synthesis netlist, therefore, unfortunately not.

**Comment by johannheyszl** [comment]
@Razer6 do I understand correctly, the issue/concern is about verification/simulation? I would relabel accordingly then.

**Comment by vogelpi** [comment]
As opposed to all other FSMs, the pwrmgr FSM doesn't have an alert output. Instead, when reaching this default (= terminal error state) everything is shutdown meaning reset and the clocks are being disabled. That's probably why it has been decided an alert signal wasn't needed.

However, pwrmgr has a fatal alert. So it would a simple and easy change to add an error output signal to the FSM which factors into the existing alert. I am happy to do this change if it's beneficial for formal FI verification. @matutem , do you think this would be okay or do you think it could cause more DV work?

**Comment by andreaskurth** [comment]
Triaged to backlog because it's a FI verification task that we don't need for V2S of pwrmgr
